### PR TITLE
Issue #3515: Add statistical rounding

### PR DESCRIPTION
**Pull request by hawkfish**
PG uses statistical (aka banker's) rounding when casting from floating point to integral types.
This triggered a number of minor expected changes,
often due to existing casts in the test output or data generation.

Disable some sqllogic tests that fail due to rounding
by comparing with a specific branch.

Update .editorconfig to stop wrapping Makefile lines.

### CODE REVIEW

File: src/common/operator/cast_operators.cpp [hunk 1]
```diff
@@ -816,7 +816,7 @@ struct IntegerCastOperation {
 		if (dbl_res < NumericLimits<result_t>::Minimum() || dbl_res > NumericLimits<result_t>::Maximum()) {
 			return false;
 		}
-		state.result = (result_t)dbl_res;
+		state.result = (result_t)std::nearbyint(dbl_res);
```
Comment (Mytherin) [inline src/common/operator/cast_operators.cpp:819]: [`std::nearbyint`](https://en.cppreference.com/w/cpp/numeric/math/nearbyint) uses global state from `std::fesetround`, which could be potentially problematic when integrating DuckDB in various clients as clients might change this setting. Is there any alternative we could explore here?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Loads of test fixes! Impressive. One minor comment:

File: src/common/operator/cast_operators.cpp (hunk 1 above)
Comment (hawkfish) [reply src/common/operator/cast_operators.cpp:819]: I'll have a look, but if it is part of the C++ runtime, I think we would be isolated.  Also, if the client wants a different setting, maybe it makes sense to let them override it?

Comment (hawkfish) [reply src/common/operator/cast_operators.cpp:819]: Looks like it is [a thread local processor setting](https://en.cppreference.com/w/cpp/numeric/fenv), but I think this is unusual enough that a if someone has changed it we should follow that setting. After all, we are an embedded acceleration engine, not a stand-alone server.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**